appvm-scripts/etc/X11/xinit/xinitrc.d/90-guivm-session.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+# Source Qubes library.
+# shellcheck disable=SC1091
+. /usr/lib/qubes/init/functions
+
+if qsvc guivm-gui-agent ;then
+    # Start Xephyr on display :1
+    DISPLAY=:0 LD_PRELOAD=/usr/lib64/qubes-gui-daemon/shmoverride.so Xephyr -ac -fullscreen -resizeable :1 &
+    # Wait for Xephyr to become ready
+    sleep 1
```

Comment:
Better use xinit. It's exact purpose is to start X server, wait for it to become ready and start another application. Handling `LD_PRELOAD` might be tricky though (similar issue to vncserver before). Probably will need modification to shmoverride library to ignore processes other than X server, somehow identified. Probably comparing `argv[0]` (shmoverride reads `/proc/self/cmdline` anyway) with a name in some environment variable.